packages/world-consumer/src/experimental/WorldConsumer.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { RESOURCE_TABLE, RESOURCE_OFFCHAIN_TABLE } from "@latticexyz/store/src/storeResourceTypes.sol";
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+import { ResourceIds } from "@latticexyz/store/src/codegen/tables/ResourceIds.sol";
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+import { ResourceAccess } from "@latticexyz/world/src/codegen/tables/ResourceAccess.sol";
+import { IBaseWorld } from "@latticexyz/world/src/codegen/interfaces/IBaseWorld.sol";
+import { WorldResourceIdLib } from "@latticexyz/world/src/WorldResourceId.sol";
+import { WorldContextConsumer } from "@latticexyz/world/src/WorldContext.sol";
+import { System } from "@latticexyz/world/src/System.sol";
+
+abstract contract WorldConsumer is System {
+  bytes14 public immutable namespace;
+
+  error WorldConsumer_RootNamespaceNotAllowed();
+  error WorldConsumer_NamespaceAlreadyExists(bytes14 namespace);
+  error WorldConsumer_NamespaceDoesNotExists(bytes14 namespace);
+  error WorldConsumer_CallerHasNoNamespaceAccess(bytes14 namespace, address caller);
+  error WorldConsumer_CallerIsNotWorld(address caller);
+  error WorldConsumer_ValueNotAllowed();
```

Comment:
I wonder if its worth adding a `worldAddress` to each? might make it easier to see what world is trying to be called if you end up configuring or calling this wrong

I also wonder about `bytes14 namespace` vs `ResourceId namespaceId` since that's generally what we pass around inside world

Reply:
ah let me check other errors

Reply:
mm so we have:

```
  error World_AccessDenied(string resource, address caller);
  error World_InvalidNamespace(bytes14 namespace);
```

As the current error is just referring to a namespace, I feel it might be more consistent to keep it as just `bytes14` but not sure.

Adding the world address makes sense.

Reply:
Added it as a `IBaseWorld world`

Reply:
I think we can keep it `address world` to simplify if you wanted to avoid casting 

I also wonder if we should use `_world()` and `WorldContext._world()`
https://github.com/latticexyz/mud/blob/b7901812d4035faa3ec9bb75f31ffe7af398bdf2/packages/world/src/WorldContext.sol#L95-L97

Instead of reimplementing + renaming as `getWorld()`?

Reply:
ah good point! will do, this was from before using world context